lib/cadence/connection/thrift.rb

```diff
@@ -345,8 +351,39 @@ def reset_sticky_task_list
         raise NotImplementedError
       end
 
-      def query_workflow
-        raise NotImplementedError
+      def query_workflow(domain:, workflow_id:, run_id:, query:, args: nil, query_reject_condition: nil)
+        request = CadenceThrift::QueryWorkflowRequest.new(
+          domain: domain,
+          execution: CadenceThrift::WorkflowExecution.new(
+            workflow_id: workflow_id,
+            run_id: run_id
+          ),
+          query: CadenceThrift::WorkflowQuery.new(
+            query_type: query,
+            query_args: args.to_json
```

Comment:
It looks like other places in this file use `JSON.serialize(args)` instead of `args.to_json`. Could you switch to that?